java-client/src/main/java/org/opensearch/client/json/JsonpSerializable.java

```diff
@@ -33,11 +33,30 @@
 package org.opensearch.client.json;
 
 import jakarta.json.stream.JsonGenerator;
+import java.io.StringWriter;
 
 /**
  * An object that is its own JsonP serializer
  */
 public interface JsonpSerializable {
 
     void serialize(JsonGenerator generator, JsonpMapper mapper);
+
+    /**
+     * A default method which returns string representation for the instances of classes
+     * implementing JsonpSerializable interface.<br>
+     * Usage : Eg for SearchRequest.class<pre>{@code SearchRequest implements JsonpSerializable{..}
+     * SearchRequest searchRequest = SearchRequest.of(request -> request...);
+     * String searchRequestString = searchRequest.writeValueAsString();}</pre>  <br>
+     *
+     */
+    default String writeValueAsString() {
+        StringWriter writer = new StringWriter();
```

Comment:
Please close the writer:

```
    try (StringWriter writer = new StringWriter()) {
        try (JsonGenerator generator = JsonpUtils.DEFAULT_PROVIDER.createGenerator(writer)) {
            serialize(generator, JsonpUtils.DEFAULT_JSONP_MAPPER);
        }

        return writer.toString();
        }
```

Reply:
StringWriter does not hold any IO resources, they maintain StringBuffer , and closing them does not accomplishes anything , it will rely on Java's garbage collection for cleanUp

Reply:
`StringWriter` implements `Closeable` and should be closed upon completion.  The fact that currently the `close()` method is noop does not mean the resource management has to be ignored.

Reply:
Okay, as you said right now close() is noop, but it does throw IO Exception , should it be an unchecked exception in which case the clients will have to handle it whenever they call the method or you think we should handle it in a specific way, any thoughts ?

Reply:
```
 try (StringWriter writer = new StringWriter()) {
        try (JsonGenerator generator = JsonpUtils.DEFAULT_PROVIDER.createGenerator(writer)) {
            serialize(generator, JsonpUtils.DEFAULT_JSONP_MAPPER);
        }

        return writer.toString();
    } catch (IOException ex) {
        thrown new UncheckedIOException(ex);
   }
```